numpy/_core/src/multiarray/convert_datatype.c

```diff
@@ -2305,6 +2322,7 @@ add_numeric_cast(PyArray_DTypeMeta *from, PyArray_DTypeMeta *to)
     if (dtypes[0]->singleton->kind == dtypes[1]->singleton->kind
             && from_itemsize == to_itemsize) {
         spec.casting = NPY_EQUIV_CASTING;
+        spec.casting |= NPY_SAME_VALUE_CASTING_FLAG;
```

Comment:
Would not be necessary if `NPY_EQUIV_CASTING` included it already.

Reply:
Can we leave this for a clean-up? It would change public-facing API (rather than appending to it).

Reply:
:+1: it might add a reasonable amount of complexity.  We could and maybe should just change `NPY_EQUIV_CASTING` (easiest/most broad solution).
However, for user-facing API it would have to NumPy runtime dependent, which is a small API break.

Now, I doubt that this matters, since I can't see anyone using `NPY_EQUIV_CASTING` in a place that needs to be a true constant (e.g. a switch statement), but in that regard I agree with splitting it out.

But it might be good to make sure we have an issue about it after merging.